main/streams/streams.c

```diff
@@ -1567,14 +1567,21 @@ PHPAPI zend_result _php_stream_copy_to_stream_ex(php_stream *src, php_stream *de
 		/* get dest open flags to check if the stream is open in append mode */
 		php_stream_parse_fopen_modes(dest->mode, &dest_open_flags);
 
+		/* Some filesystems will cause failures if the max length is greater than the file length.
+		 * We therefore use a stat call to get the actual file size. Since stat calls are cached
+		 * this shouldn't have much impact. */
+		php_stream_statbuf statbuf;
```

Comment:
> Since stat calls are cached this shouldn't have much impact

This can be an issue when the cached size is higher than the actual size. There could be some race conditions as well, if the file is changed after the stat call.

What errno do we get when copy_file_range() fails due to a too large size? Can we fallback to other methods?

Reply:
Ah, that's unfortunate.
It's errno EIO which is one of thosr catch-all errors. I'll just do a fallback when the error is EIO

Reply:
I reported the bug here #10548 - strace shows:
copy_file_range(5, NULL, 6, NULL, 9223372036854775807, 0) = -1 EIO (Input/output error)
The current code gives up on EIO since it is the default: /* unexpected I/O error - give up, no fallback */.